### PR TITLE
Refactor of CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by bramceulemans**
### What is the reason for this PR?

- [x] A new feature
- [ ] Fixes issue #

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [x] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

The contribution guide was quite outdates, I have updated links and rewritten parts that refer to documenting in the `README.md`.

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
+* If your new Formatter is specific to a certain locale, document them in the respective [locale documentation](https://github.com/FakerPHP/fakerphp.github.io/tree/main/docs/locales).
+* Avoid changing existing sets of data. Some developers use Faker for seeding unit tests, changing the data makes their tests fail.
+* Speed is important to Faker. Make sure your code is optimized to generate thousands of fake items in no time, without consuming too much CPU or memory.
+* If you commit a new feature, be prepared to help maintaining it. Watch the project on GitHub, and please comment on issues or pull requests regarding the feature you contributed.
 
-Once your code is merged, it is available for free to everybody under the MIT License. Publishing your Pull Request on the Faker GitHub repository means that you agree with this license for your contribution.
+Once your code is merged, it is available for free to everybody under the [MIT License](https://github.com/FakerPHP/Faker/blob/main/LICENSE). Publishing your pull request on the Faker GitHub repository means that you agree with this license for your contribution.
```
Comment (GrahamCampbell) [inline .github/CONTRIBUTING.md:19]: Just write `[MIT License](LICENSE)`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -16,6 +16,6 @@ Before proposing a pull request, check the following:
 * Speed is important to Faker. Make sure your code is optimized to generate thousands of fake items in no time, without consuming too much CPU or memory.
 * If you commit a new feature, be prepared to help maintaining it. Watch the project on GitHub, and please comment on issues or pull requests regarding the feature you contributed.
 
-Once your code is merged, it is available for free to everybody under the [MIT License](https://github.com/FakerPHP/Faker/blob/main/LICENSE). Publishing your pull request on the Faker GitHub repository means that you agree with this license for your contribution.
+Once your code is merged, it is available for free to everybody under the [MIT License](/LICENSE). Publishing your pull request on the Faker GitHub repository means that you agree with this license for your contribution.
```
Comment (GrahamCampbell) [inline .github/CONTRIBUTING.md:19]: Without the slash.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Something seems up with the national insurance stuff. Either we have flakey generation, or a flakey test?

**Comment by bramceulemans** [comment]
I have no clue, as you can see nothing regarding that has changed and it's passing on main.

**Comment by bramceulemans** [comment]
I found the cause, it's not the bank account numbers, it's the new test for `en_GB::nino`:

```
1) Faker\Test\Provider\en_GB\PersonTest::testNationalInsuranceNumber
Failed asserting that true is false.
```

**Comment by pimjansen** [comment]
I raised a bug for this to review since it is not consistent it seems

**Comment by bramceulemans** [comment]
Issue with the test is fixed in main. @GrahamCampbell

**Pull request merged**